packages/frontend-core/src/api/index.ts

```diff
@@ -163,7 +163,13 @@ export const createAPIClient = (config: APIClientConfig = {}): APIClient => {
         } else if (parseResponse) {
           return await parseResponse(response)
         } else {
-          return (await response.json()) as ResponseT
+          const text = await response.text()
+          // If the response has no body at all e.g s3.put
+          if (!text) {
+            // Empty response
+            return {} as ResponseT
+          }
+          return JSON.parse(text) as ResponseT
```

Comment:
Rather than handle this here and kind of lie to TS, we could add a `parseResponse` property to the API PUT call for the external upload. That's the proper way of handling a unique type of response. In this case, if we don't care about the response we can add `parseResponse: () => {}` or equivalent. That would go around line 103 in `attachments.ts`. Just think that's a little cleaner than assuming any empty response is a valid `ResponseT`.

Reply:
Wayyyyy neater approach. I've reverted the api change and added in a `parseResponse` exclusively for the externalUpload. Thanks!